google/resource_google_organization_policy.go

```diff
@@ -0,0 +1,327 @@
+}
+
+func expandBooleanOrganizationPolicy(configured []interface{}) *cloudresourcemanager.BooleanPolicy {
+	if len(configured) == 0 {
+		return nil
+	}
+
+	booleanPolicy := configured[0].(map[string]interface{})
+	return &cloudresourcemanager.BooleanPolicy{
+		Enforced: booleanPolicy["enforced"].(bool),
+	}
+}
+
+func flattenListOrganizationPolicy(policy *cloudresourcemanager.ListPolicy) []map[string]interface{} {
+	lPolicies := make([]map[string]interface{}, 0, 1)
+
+	if policy == nil {
+		return lPolicies
+	}
+
+	listPolicy := map[string]interface{}{}
+	switch {
+	case policy.AllValues == "ALLOW":
+		listPolicy["allow"] = []interface{}{map[string]interface{}{
+			"all": true,
+		}}
+	case policy.AllValues == "DENY":
+		listPolicy["deny"] = []interface{}{map[string]interface{}{
+			"all": true,
+		}}
+	case len(policy.AllowedValues) > 0:
+		listPolicy["allow"] = []interface{}{map[string]interface{}{
+			"values": schema.NewSet(schema.HashString, convertStringArrToInterface(policy.AllowedValues)),
+		}}
+	case len(policy.DeniedValues) > 0:
+		listPolicy["deny"] = []interface{}{map[string]interface{}{
+			"values": schema.NewSet(schema.HashString, convertStringArrToInterface(policy.DeniedValues)),
+		}}
+	}
+
+	lPolicies = append(lPolicies, listPolicy)
+
+	return lPolicies
+}
+
+func expandListOrganizationPolicy(configured []interface{}) (*cloudresourcemanager.ListPolicy, error) {
+	if len(configured) == 0 {
+		return nil, nil
+	}
+
+	listPolicyMap := configured[0].(map[string]interface{})
+
+	allow := listPolicyMap["allow"].([]interface{})
+	deny := listPolicyMap["deny"].([]interface{})
+
+	if len(allow) > 0 && len(deny) > 0 {
+		return nil, fmt.Errorf("You cannot specified an `allow` and a `deny` block")
+	}
+
+	var allValues string
+	var allowedValues []string
+	var deniedValues []string
+	if len(allow) > 0 {
+		allowMap := allow[0].(map[string]interface{})
+		all := allowMap["all"].(bool)
+		values := allowMap["values"].(*schema.Set)
+
+		if all && values.Len() > 0 {
+			return nil, fmt.Errorf("Invalid `allow` block, `all` cannot be used in conjunction with a specific list of `values`")
+		}
+
+		if all {
+			allValues = "ALLOW"
+		} else {
+			allowedValues = convertStringArr(values.List())
+		}
+	}
+
+	if len(deny) > 0 {
+		denyMap := deny[0].(map[string]interface{})
+		all := denyMap["all"].(bool)
+		values := denyMap["values"].(*schema.Set)
+
+		if all && values.Len() > 0 {
+			return nil, fmt.Errorf("Invalid `deny` block, `all` cannot be used in conjunction with a specific list of `values`")
+		}
+
+		if all {
+			allValues = "DENY"
+		} else {
+			deniedValues = convertStringArr(values.List())
+		}
+	}
+
+	listPolicy := configured[0].(map[string]interface{})
+	return &cloudresourcemanager.ListPolicy{
+		AllValues:      allValues,
+		AllowedValues:  allowedValues,
+		DeniedValues:   deniedValues,
+		SuggestedValue: listPolicy["suggested_value"].(string),
+	}, nil
+}
+
+func canonicalOrgPolicyConstraint(constraint string) string {
```

Comment:
if we allow people to set a non-canonical version, wouldn't that lead to diffs next plan once the constraint comes back from the api with a canonical version? Should be an easy enough fix with a diffsuppress

Reply:
There is already a `DiffSuppress` func set in the schema.